recipes-bsp/u-boot/u-boot-starfive_v2021.10.bb

```diff
@@ -0,0 +1,42 @@
+require recipes-bsp/u-boot/u-boot-common.inc
+require recipes-bsp/u-boot/u-boot.inc
+
+LIC_FILES_CHKSUM = "file://Licenses/README;md5=5a7450c57ffe5ae63fd732446b988025"
+
+FILESEXTRAPATHS:prepend := "${THISDIR}/files:"
+
+BRANCH:visionfive2 = "JH7110_VisionFive2_devel"
+
+SRC_URI = "git://github.com/starfive-tech/u-boot.git;protocol=https;branch=${BRANCH} \
+           file://tftp-mmc-boot.txt \
+          "
+
+SRC_URI:append:visionfive2 = " \
+        file://fix-riscv-isa.patch \
+        file://uEnv-visionfive2.txt \
+"
+
+# tag VF2_v2.10.4
+SRCREV:visionfive2 = "1ae835c25e3d8b7bc3c8981fb720f030c1892d57"
+
+DEPENDS:append = " u-boot-tools-native"
+
+# Overwrite this for your server
+TFTP_SERVER_IP ?= "127.0.0.1"
+
+do_configure:prepend() {
+    sed -i -e 's,@SERVERIP@,${TFTP_SERVER_IP},g' ${WORKDIR}/tftp-mmc-boot.txt
+    mkimage -O linux -T script -C none -n "U-Boot boot script" \
+        -d ${WORKDIR}/tftp-mmc-boot.txt ${WORKDIR}/${UBOOT_ENV_BINARY}
+}
+
+do_deploy:append:visionfive2() {
+    install -m 644 ${WORKDIR}/uEnv-visionfive2.txt ${DEPLOYDIR}/uEnv.txt
+}
+
+COMPATIBLE_MACHINE = "(visionfive2)"
```

Comment:
does it support `visionfive1` as well? if so can we add that here as well, so we can have both machines share same u-boot.

Reply:
JH7110_VisionFive2_devel branch of [github.com/starfive-tech/u-boot](https://github.com/starfive-tech/u-boot) moved to U-Boot v2021.10. Seems like this particular branch/tag is missing `visionfive1` support. I decided to create a new recipe because all branches supporting `visionfive1` are based on v2021.04.

Reply:
hmmm, this is not good, Can you check with upstream if there is a plan to maintain single branch for all visionfive variants ?

Reply:
Asked here https://github.com/starfive-tech/u-boot/issues/43.

JH7110/Visionfive2 support in mainline U-Boot is currently under review (see https://rvspace.org/en/project/JH7110_Upstream_Plan and https://patchwork.ozlabs.org/project/uboot/list/?series=346514), so hopefully we can use upstream recipe soon.

Meanwhile, current u-boot-starfive_v2021.04 in meta-riscv is kind of broken:
* BRANCH for Visionfive is "JH7100_VisionFive_OH_dev" which is based on v2022.04
* BRANCH for Visionfive2  is "JH7110_VisionFive2_devel" which is based on v2021.10
* BRANCH for beaglev-starlight-jh7100 is "Fedora_JH7100_2021.04" which is based on  v2021.04

@kraj I guess the optimal solution now would be to rename "u-boot-starfive_v2021.04" to "u-boot-starfive" and set branches/SRCREV for each board.